modules/spring-config/src/main/java/org/mule/runtime/config/spring/dsl/model/MinimalApplicationModelGenerator.java

```diff
@@ -88,17 +93,43 @@ public ApplicationModel getMinimalModelByName(String name) {
 
   private ComponentModel filterFlowModelParts(ComponentModel flowModel, String[] parts) {
     ComponentModel currentLevelModel = flowModel;
+    // TODO MULE-11482: improve this ugly code when new flow path mechanism is in place
     for (int i = 1; i < parts.length; i++) {
-      int selectedPath = Integer.parseInt(parts[i]);
-      List<ComponentModel> innerComponents = currentLevelModel.getInnerComponents();
-      Iterator<ComponentModel> iterator = innerComponents.iterator();
-      int currentElement = 0;
-      while (iterator.hasNext()) {
-        if (currentElement != selectedPath) {
+      String part = parts[i];
+      switch (part) {
+        case "source": {
+          ComponentModel sourceComponentModel = flowModel.getInnerComponents().get(0);
+          checkState(areMatchingTypes(MessageSource.class, sourceComponentModel.getType()),
+                     format(
+                            "Flow path is pointing to the message source of flow %s but it seems that the flow does not have a message source, component identifier of first element is %s",
+                            flowModel.getNameAttribute(), sourceComponentModel.getIdentifier()));
+          Iterator<ComponentModel> iterator = flowModel.getInnerComponents().iterator();
           iterator.next();
```

Comment:
Maybe a comment on this line would help why the next is being done before the hasNext.